### PR TITLE
fix(docs): update core-tutorial

### PR DESCRIPTION
**Pull request by dhmlau**
Update the core tutorial in `core-tutorial`.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/tutorials/core/3-context-in-action.md [hunk 1]
```diff
@@ -55,37 +61,86 @@ export class GreetingApplication extends BootMixin(RestApplication) {
...
+   The constant value can be a string, number or any other types.
 
-4. To be created from a provider class
+2. To be created by a factory function.
+
+   For example, a function to get the current time.
+
+3. To be instantiated from a class.
+
+   A class can be aservice, controller or other LoopBack artifacts.
```
Comment (raymondfeng) [inline docs/site/tutorials/core/3-context-in-action.md]: a service

File: docs/site/tutorials/core/3-context-in-action.md [hunk 2]
```diff
@@ -55,37 +61,86 @@ export class GreetingApplication extends BootMixin(RestApplication) {
...
 
-1. Transient
-2. Singleton
-3. Context
+There are three binding scopes:
+
+1. Transient - a new instance of binding is created for each request.
+2. Singleton - there is only a single instance of the binding.
+3. Context - for a given context in the hierarchy, there is only a single
+   instance for a given binding key.
```
Comment (raymondfeng) [inline docs/site/tutorials/core/3-context-in-action.md]: Add a link to https://loopback.io/doc/en/lb4/Binding.html#configure-the-scope

File: docs/site/tutorials/core/3-context-in-action.md [hunk 3]
```diff
@@ -55,37 +61,86 @@ export class GreetingApplication extends BootMixin(RestApplication) {
...
 
-https://github.com/strongloop/loopback-next/blob/master/examples/context/src/context-observation.ts
+The context view gives us control when an binding is being added or moreved.
+This also allows the support of dynamic extension points. For the
+[greeting-app example](https://github.com/strongloop/loopback-next/tree/master/examples/greeting-app),
+after the application has started, more greeters can be added, and there is no
+need to add all the greeters up front.
+
+To see an example, go to:
+https://github.com/strongloop/loopback-next/blob/master/examples/context/src/context-observation.ts.
```
Comment (raymondfeng) [inline docs/site/tutorials/core/3-context-in-action.md]: Add a link to https://loopback.io/doc/en/lb4/Context.html#context-observers

**Comment by dhmlau** [comment]
The 2 failures in CI are not related to my changes.  It's probably because the branch is out of date with master. 
```
1) lb4 example
       accepts the example name via interactive prompt:
      AssertionError [ERR_ASSERTION]: '1.8.3' == '1.8.0'
      + expected - actual
      -1.8.3
      +1.8.0
      
      at Object.keys.forEach.key (packages\cli\node_modules\yeoman-assert\index.js:235:12)
      at Array.forEach (<anonymous>)
      at Function.assert.objectContent (packages\cli\node_modules\yeoman-assert\index.js:229:24)
      at Function.assert.JSONFileContent.assert.jsonFileContent (packages\cli\node_modules\yeoman-assert\index.js:263:10)
      at testUtils.executeGenerator.withPrompts.then (packages\cli\test\integration\generators\example.integration.js:55:16)
      at <anonymous>
  2) lb4 example
       accepts the example name as a CLI argument:
      AssertionError [ERR_ASSERTION]: '1.8.3' == '1.8.0'
      + expected - actual
      -1.8.3
      +1.8.0
      
      at Object.keys.forEach.key (packages\cli\node_modules\yeoman-assert\index.js:235:12)
      at Array.forEach (<anonymous>)
      at Function.assert.objectContent (packages\cli\node_modules\yeoman-assert\index.js:229:24)
      at Function.assert.JSONFileContent.assert.jsonFileContent (packages\cli\node_modules\yeoman-assert\index.js:263:10)
      at testUtils.executeGenerator.withArguments.then (packages\cli\test\integration\generators\example.integration.js:70:16)
      at <anonymous>
```

**Pull request merged**